### PR TITLE
EOS-23626 : Create support.yaml for HA

### PR DESCRIPTION
**Pull request by s-arya**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
EOS-23626 : Create support.yaml for HA
  </code>
</pre>
## Unit testing on RPM done
<pre>
Yes, on single node and 3 node VM. Refer unit cases section for tests.
  
</pre>
## Problem Description
<pre>
  <code>
 With addition of upgrade related entries in setup.yaml, YAML parsing of setup.yaml is broken (but provisioner is capable of handling this).
Command for component specific support bundle generation is in setup.yaml
Due to broken parsing, support bundle generation is at risk and a quick remedy is required.
  </code>
</pre>
## Solution
    -Create support.yaml file which will have the "support_bundle" entry from setup.yaml 
    -Install this support.yaml file at the same location where current setup.yaml file resides after installation.

## Unit Test Cases

-  Automated single node deployment passed : http://eos-jenkins.colo.seagate.com/job/Cortx-Deployment/job/VM-Deployment-R2-1Node/1866/
- Automated 3 node deployment passed : http://eos-jenkins.colo.seagate.com/job/Cortx-Deployment/job/VM-Deployment-R2-3Node/754/
- RPM testing  - details below
<pre>
<code>
srvnode-1
----------
[root@ssc-vm-g2-rhev4-0626 ~]# rpm -qa | grep cortx-ha-
cortx-ha-2.0.0-2504_7e8381a.x86_64

[root@ssc-vm-g2-rhev4-0626 ~]# find /opt/seagate/cortx/ha/ -name *.yaml
/opt/seagate/cortx/ha/conf/setup.yaml
/opt/seagate/cortx/ha/conf/support.yaml

srvnode-2
----------
root@ssc-vm-g2-rhev4-0627 ~]# rpm -qa | grep cortx-ha-
cortx-ha-2.0.0-2504_7e8381a.x86_64

[root@ssc-vm-g2-rhev4-0627 ~]# find /opt/seagate/cortx/ha/ -name *.yaml
/opt/seagate/cortx/ha/conf/setup.yaml
/opt/seagate/cortx/ha/conf/support.yaml

srvnode-3 
---------
[root@ssc-vm-g2-rhev4-0628 ~]#  rpm -qa | grep cortx-ha-
cortx-ha-2.0.0-2504_7e8381a.x86_64

[root@ssc-vm-g2-rhev4-0628 ~]# find /opt/seagate/cortx/ha/ -name *.yaml
/opt/seagate/cortx/ha/conf/setup.yaml
/opt/seagate/cortx/ha/conf/support.yaml
  </code>
 </pre>

### CODE REVIEW

**Review by ajay-paratmandali — APPROVED** [review]

**Comment by ajay-paratmandali** [comment]
@sachinpunadikar Please review

**Review by indrajitzagade — APPROVED** [review]

File: jenkins/build_with_pyinstaller.sh [hunk 1]
```diff
@@ -148,6 +148,7 @@ then
      cp -rf $BASE_DIR/conf/script/v1/* $HA_DIR/conf/script/
      cp -rf $BASE_DIR/conf/iostack-ha/ $HA_DIR/conf/
      cp -rf $BASE_DIR/conf/mini_provisioner/v1/setup.yaml $HA_DIR/conf/setup.yaml
+     cp -rf $BASE_DIR/conf/mini_provisioner/v1/support.yaml $HA_DIR/conf/support.yaml
```
Comment (ArchanaLimaye) [inline jenkins/build_with_pyinstaller.sh:151]: This folder name needs to be v2 right ?

File: conf/mini_provisioner/v2/support.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+support_bundle:
+    - cortx support_bundle create
```
Comment (sachinpunadikar) [inline conf/mini_provisioner/v2/support.yaml:17]: Wondering whether "cortx" is the command ? If so will it create support bundle for HA ?
In general every component has a script name having the component name it. For example, below is from S3:
support_bundle:
  - sh /opt/seagate/cortx/s3/scripts/s3_bundle_generate.sh

Comment (s-arya) [reply conf/mini_provisioner/v2/support.yaml:17]: The "cortx" is a CLI which is owned by HA and installed as part of rpm. Currently, generating support bundle for HA is done using the cortx cli, by passing above arguments to that cli.

File: jenkins/build_with_pyinstaller.sh (hunk 1 above)
Comment (s-arya) [reply jenkins/build_with_pyinstaller.sh]: Good catch. Fixed it in latest commit.

**Comment by s-arya** [comment]
Updated the patch and re-ran all the unit tests in the "Unit tests section"
3-node deployment passed with latest build : http://eos-jenkins.colo.seagate.com/job/Cortx-Deployment/job/VM-Deployment-R2-3Node/760/

File: conf/mini_provisioner/v2/support.yaml (hunk 2 above)
Comment (sachinpunadikar) [reply conf/mini_provisioner/v2/support.yaml:17]: Thanks for the explanation

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by ArchanaLimaye — APPROVED** [review]

**Pull request merged**